railib/api.py

```diff
@@ -221,11 +227,15 @@ def _get_resource(ctx: Context, path: str, key=None, **kwargs) -> Dict:
     url = _mkurl(ctx, path)
     rsp = rest.get(ctx, url, **kwargs)
     rsp = json.loads(rsp.read())
+
     if key:
         rsp = rsp[key]
-    if rsp and isinstance(rsp, list):
-        assert len(rsp) == 1
+
+    if isinstance(rsp, list):
+        if len(rsp) == 0:
+            raise ResourceNotFoundError(f"Resource not found at {url}")
```

Comment:
is it correct to say that historically an assertion error would be thrown instead of ResourceNotFound?  (If so its good, just want to make sure we minor-version bump since surface area is changing)

Reply:
Nope, the assert was there to check if for some reason we get two resources with the same name. I don't think it was useful since the name is required to be unique for any resource. But even if we do get two resources with the same name for some reason the assert wouldn't do any good for the user as the user would just get confused.

The change is backward incompatible, previously the function would return an empty list if a requested resource not found. So any method like get_engine or get_database would return an empty list when we can't find a resource with the provided name. Ideally, our rest api should handle this and give 404 but since we don't have a rest api for it I emulate this behavior by throwing ResourceNotFoundError.